pkg/controller/account/byoc.go

```diff
@@ -79,16 +77,14 @@ func (r *ReconcileAccount) initializeNewCCSAccount(reqLogger logr.Logger, accoun
 	}
 	if clientErr != nil {
 		utils.SetAccountClaimStatus(
+			r.Client,
+			reqLogger,
 			accountClaim,
 			"Failed to create AWS Client",
 			"AWSClientCreationFailed",
 			awsv1alpha1.ClientError,
 			awsv1alpha1.ClaimStatusError,
 		)
-		err := r.Client.Status().Update(context.TODO(), accountClaim)
-		if err != nil {
-			reqLogger.Error(err, "Failed to create AWS Client")
-		}
 
 		if accountClaim != nil {
 			claimErr := r.setAccountClaimError(reqLogger, account, clientErr.Error())
```

Comment:
If someone can comment - Can this be removed as well? We've technically just set the AccountClaim to an error state using the call to `SetAccountClaimStatus`, which is what setAccountClaimError calls as well